source/ranges.tex

```diff
@@ -10040,3 +10070,1603 @@
+for use on subsequent calls
+when \tcode{V} models \exposconcept{slide-caches-first}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr auto begin() const requires @\exposconcept{slide-caches-nothing}@<const V>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{\exposid{iterator}<true>(ranges::begin(\exposid{base_}), \exposid{n_})}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr auto end()
+  requires (!(@\exposconcept{simple-view}@<V> && @\exposconcept{slide-caches-nothing}@<const V>));
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\begin{itemize}
+\item
+If \tcode{V} models \exposconcept{slide-caches-nothing},
+\begin{codeblock}
+@\exposid{iterator}@<false>(ranges::begin(@\exposid{base_}@) + range_difference_t<V>(size()), @\exposid{n_}@)
+\end{codeblock}
+\item
+Otherwise, if \tcode{V} models \exposconcept{slide-caches-last},
+\begin{codeblock}
+@\exposid{iterator}@<false>(ranges::prev(ranges::end(@\exposid{base_}@), @\exposid{n_}@ - 1, ranges::begin(@\exposid{base_}@)), @\exposid{n_}@)
+\end{codeblock}
+\item
+Otherwise, if \tcode{V} models \libconcept{common_range},
+\begin{codeblock}
+@\exposid{iterator}@<false>(ranges::end(@\exposid{base_}@), ranges::end(@\exposid{base_}@), @\exposid{n_}@)
+\end{codeblock}
+\item
+Otherwise, \tcode{\exposid{sentinel}(ranges::end(\exposid{base_})).}
+\end{itemize}
+
+\pnum
+\remarks
+In order to provide the amortized constant-time complexity
+required by the \libconcept{range} concept,
+this function caches the result within the \tcode{slide_view}
+for use on subsequent calls
+when \tcode{V} models \exposconcept{slide-caches-last}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr auto end() const requires slide-caches-nothing<const V>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{begin() + range_difference_t<const V>(size())}.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr auto size() requires sized_range<V>;
+constexpr auto size() const requires sized_range<const V>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+auto sz = ranges::distance(@\exposid{base_}@) - @\exposid{n_}@ + 1;
+if (sz < 0) sz = 0;
+return @\exposid{to-unsigned-like}@(sz);
+\end{codeblock}
+\end{itemdescr}
+
+\rSec3[range.slide.iterator]{Class template \tcode{slide_view::\exposid{iterator}}}
+
+\indexlibraryglobal{slide_view::iterator}%
+\begin{codeblock}
+namespace std::ranges {
+  template<@\libconcept{forward_range}@ V>
+    requires @\libconcept{view}@<V>
+  template<bool Const>
+  class slide_view<V>::@\exposid{iterator}@ {
+    using @\exposid{Base}@ = @\exposid{maybe-const}@<Const, V>;                 // \expos
+    iterator_t<@\exposid{Base}@> @\exposid{current_}@   = iterator_t<@\exposid{Base}@>();   // \expos
+    iterator_t<@\exposid{Base}@> @\exposid{last_ele_}@  = iterator_t<@\exposid{Base}@>();   // \expos,
+                                               // present only if \tcode{\exposid{Base}} models \tcode{\exposconcept{slide-caches-first}}
```

Comment:
```suggestion
                                               // present only if \exposid{Base} models \exposconcept{slide-caches-first}
```

Reply:
The font isn't quiite right otherwise, I think.

Reply:
Perhaps that's worth a separate editorial issue. My suggestion seems to be the norm, where macros that internally use `\tcode` are not redundantly wrapped in `\tcode`.

Reply:
I meant not an editorial issue for my suggestion here, but for adding `\tcode` everywhere within comments, even if already implied. I leave that to you.